File: tournaments/isSumOfConsecutive2/isSumOfConsecutive2.cpp
int isSumOfConsecutive2(int n) {
  int result = 0;
  for (int start = 1; start < n; start++) {
    int number = n,
        subtrahend = start;
    while (number > 0) {
      number -= subtrahend;
      subtrahend++;
    }
    if (!number) {
      result++;
    }
  }
  return result;
}

File: tournaments/isSumOfConsecutive2/isSumOfConsecutive2.js
function isSumOfConsecutive2(n) {
    let result = 0;

    for (let i = 1; n > i; ++i) {
        let a = 0;
        let b = i;

        while (n > a) a += b++;
        result += n === a;
    }

    return result;
}
